backend/routes/api/resources.js

```diff
@@ -0,0 +1,248 @@
+const express = require("express");
+const router = express.Router();
+const Resource = require("../../models/Resource");
+const { celebrate, Joi } = require("celebrate");
+const geolib = require("geolib");
+const Fuse = require("fuse.js");
+Joi.objectId = require("joi-objectid")(Joi);
+
+// get all resources
+router.get("/", async (req, res) => {
+  const resources = await Resource.find({});
+  res.json({
+    code: 200,
+    result: resources,
+    success: true
+  });
+});
+
+// get list of resources filtered by location radius
+router.get(
+  "/filter",
+  celebrate({
+    query: {
+      radius: Joi.number(),
+      lat: Joi.number(),
+      long: Joi.number(),
+      keyword: Joi.string()
+    }
+  }),
+  async (req, res) => {
+    const radius = req.query.radius;
+    const lat = req.query.lat;
+    const long = req.query.long;
+    const keyword = req.query.keyword;
+
+    let resources = await Resource.find({});
+
+    // 3963.2 = radius of Earth in miles
+    if (radius && lat && long) {
+      resources = await Resource.find({
+        location: {
+          $geoWithin: { $centerSphere: [[long, lat], radius / 3963.2] }
+        }
+      });
+
+      // sort by closest distance first
+      resources.sort(function(a, b) {
+        const distanceA = geolib.getDistance(
+          {
+            latitude: a.location.coordinates[0],
+            longitude: a.location.coordinates[1]
+          },
+          { latitude: lat, longitude: long }
+        );
+        const distanceB = geolib.getDistance(
+          {
+            latitude: b.location.coordinates[0],
+            longitude: b.location.coordinates[1]
+          },
+          { latitude: lat, longitude: long }
+        );
+
+        if (distanceA < distanceB) {
+          return -1;
+        }
+        return 1;
+      });
+    }
+    if (keyword) {
+      // fuzzy search
+      var options = {
```

Comment:
Let's extract this to another function that does fuzzy searching so we have single responsibility and modularity here (we may get resources another way and want to fuzzy search on it as well)